pentaho-osgi-utils-impl/src/main/java/org/pentaho/osgi/impl/KarafFeatureWatcherImpl.java

```diff
@@ -49,6 +50,14 @@ public KarafFeatureWatcherImpl( BundleContext bundleContext ) {
 
     long entryTime = System.currentTimeMillis();
 
+    ServiceTracker serviceTracker = new ServiceTracker( bundleContext, FeaturesService.class.getName(), null );
+    serviceTracker.open();
+    try {
+      serviceTracker.waitForService( timeout );
+    } catch ( InterruptedException e ) {
```

Comment:
Where are you getting the IKarafBlueprintWatcher and calling waitForBlueprint() ? shouldn't it be around here?